Update tax-use-cases.md (#388)
AdobeDocs/commerce-extensibility: src/pages/starter-kit/checkout/tax-use-cases.md

  }
}
```

> ## Propagate serialized custom attributes

Reviewer: Maybe `Propagate custom attributes of tax classes` shows the purpose better? Because the tax module provides propagating custom attributes from tax classes rather than just custom attributes from the oope-custom-attributes module side.

### Tax class custom attributes available in Quote

> The `oope-tax-management` module introduces support for assigning serialized custom attributes to tax classes. These attributes are then automatically associated with shopping cart data during cart creation and product addition. This allows for tax-related metadata to be included early in the checkout process and carried forward into subsequent operations.

Reviewer: `oope-tax-management` is not a module name, repo name.
`magento/module-out-of-process-tax-management` is the composer module name: https://developer.adobe.com/commerce/extensibility/starter-kit/checkout/tax-install/#installation

However, I'd just write down abstractly such as out-of-process tax module as the previous section does. Also, I think some content are pretty overlap with the previous section in this page, look: https://developer.adobe.com/commerce/extensibility/starter-kit/checkout/tax-use-cases/#update-custom-attributes-on-tax-classes-via-admin-ui
> The out-of-process tax module allows you to add custom attributes to tax classes. These attributes are useful when integrating with third-party tax systems that require standardized identifiers or additional metadata.
For the relevant endpoints to update tax class custom attributes, see the [Tax API reference](https://developer.adobe.com/commerce/extensibility/starter-kit/checkout/tax-reference/).

You can consider making the previous section connected for the reader, such as the custom attributes available to the tax (link to the previous section), are then automatically associate~

> ### How it works

Reviewer: I'd remove this section title as it works without it and for aesthetic purpose of index appearing in the page (as it's too long now).

- **Customer Tax Class → Quote**:  
  When a customer creates a shopping cart, any serialized custom attributes associated with the customer's tax class are copied to the `Quote` entity.

- **Product Tax Class → Quote Item**:  
  When a product is added to the cart, serialized custom attributes from the product's tax class are copied to the corresponding `Quote Item`.

This ensures that both the `Quote` and each `Quote Item` contain tax-relevant custom data for further processing or integration.

> ---

Reviewer: I don't see any reference to using `---` in this repository, so I would suggest removing it. 
Generally, when writing in a specific style or sentences, I recommend reviewing the existing documents and aligning with their writing style, as writers are likely to request changes otherwise.

### Quote to Order propagation

Once the customer places an order, all serialized custom attributes already present in the `Quote` and `Quote Item` entities are automatically propagated to the `Order` and `Order Item` entities, respectively.

> > This behavior was implemented previously as part of the custom attributes module and does not require additional setup from the tax module.

Author: This quotation basically pointing to specify  that "The copy of serialized custom attributes from Quote and Quote Item to Order and Order Item was already implemented in the new custom attributes module."

The purpose of this propagation is to ensure that tax class metadata—initially attached via customer and product associations—is consistently preserved throughout the entire checkout lifecycle. This allows external systems (such as tax calculation services or order analytics platforms) to access the tax class and its serialized custom attributes via:

- REST APIs
- GraphQL queries

> - Event-driven integrations (via Adobe Commerce Eventing)

Reviewer: @MohsinZamanShaheen I've seen that @nofuss shared comments that they are available in the following apis: https://github.com/AdobeDocs/commerce-extensibility/pull/388/files#r2131805277
> https://github.com/AdobeDocs/commerce-extensibility/pull/388#issuecomment-2958046341
> Regarding the retrieving of this information via graphql, rest or eventing. I mentioned this availability and all three option available but in terms of adding specific examples so I'll need to have a look at that and find out as I never used that.

My small concern is that I usually try to verify the behavior before writing in the official documentation, but if Russ/Sasha confirm that these are working, then ok. I guess GraphQL or REST work as attached in the custom attribute's guide, not sure about Eventing, maybe custom attributes documentation can mention about Eventing too: https://developer.adobe.com/commerce/services/cloud/guides/custom-attributes/#query. Next time please let us know if you need any help so that we can check it together earlier.

Author: @kuflower I agree with you, it's ideal to verify before but unfortunately didn't knew about the different options before. I'll try to verify but GraphQL and REST works for sure. Eventing I'll check with Russ or Sasha to make sure before publishing.

  }
}
```

> ## Propagate serialized custom attributes

Reviewer 2: As for the purpose of this feature, both tax classes and their custom attributes will be stored with the Quote plus Quote Item entities and ultimately the Order and Order Item entities, and available to external systems via Rest, GraphQL or Eventing.

  }
}
```

> ## Propagation of serialized custom attributes from Tax Class entities to Quote/QuoteItem entities
> 
> This new feature populates the `custom_attributes` fields of the `Quote` and `Quote_item` entities as following:

Reviewer 3: ```suggestion
## Propagate serialized custom attributes

The following example describes how you can populate  the `custom_attributes` fields of the `Quote` and `Quote_item` entities by using custom attributes.
```

- `Quote`: Any serialized custom attributes from the customer tax class are populated into the `custom_attributes` field of this `Quote` object.
- `Quote_item`: Any serialized custom attributes from the tax class of the product assocaited to an item` are set as `custom_attributes` of this `Quote_item` object.

Once `custom_attributes` are populated, these are subsequently propagated to the `Order` and `Order_item` entities as per correspondance.

> In the case of `Quote_item` currently only serialized custom attributes from the tax class of the `product`are being propagated but there are other types of items as well such as `shipping`or `giftwrap` which will be considered for future enhancements.

Author: I know this might look extra but we actually want to explicitly mention `shipping`or `giftwrap` items since it's a gap that will be considered in future.

### Example

Here's an example of how this looks like in case of a `Quote_item` but it's similar for 'quote':

> If the following is the tax class information where we'll mainly focus on the `custom_attributes_serializable` field for now:

Reviewer 3: I don't understand this. Replace this with a short simple description of what the example does and how the user would use it. Where do they add the custom attributes? Is this this compatible with PaaS and SaaS? etc

```json
{

> "class_id":"2","class_name":"Taxable Goods UPDATED","class_type":"PRODUCT","custom_attributes_serializable":"{\"itemProduct_code\":\"item_product_code_updated\",\"product_code_2\":\"product_quote_item_code_2\"}"

Author: @kuflower For now I think actually the example might not be necessary. I rewrote everything and I think it conveys the feature's aspects clearly so maybe for now we can skip this.

}

> 

Reviewer 3: ```suggestion
```

```
 

> Any attributes from the field will be propagated to the `Quote_item` object as shows in the following simplified version of a `quote_item` object:

Reviewer 3: "the field" is ambiguous. Respecify the field name. Don't use future tense.

```json
{
"quote_id: 23 created_at: 2025-05-30 11:14:29 updated_at: 2025-06-04 09:07:24 product_id: 3 store_id: 1 parent_item_id: NULL is_virtual: 0 sku: 24-MB03 free_shipping: 0 "custom_attributes_serializable": {"itemProduct_code": "item_product_code_updated","product_code_2": "product_quote_item_code_2"}

> }

Reviewer 3: The first half of this example doesn't have quotes and the quote in front of `quote_id` is never closed.

```

> And subsequently this information will be propagate to `Order` and `Order_item` level once the order is placed.

Reviewer 3: ```suggestion
Once an order is placed, this information will propagate to the `Order` and `Order_item` level.
```

> 
> 
> 

Reviewer 3: ```suggestion
```
